### PR TITLE
feat(menu): migrate to spectrum-tokens + add new features

### PR DESCRIPTION
**Pull request by pfulton**
## Description

- Migrates the **menu** component to use core tokens.
- Adds new features / variants.
- Adds new stories and examples to cover new variants and some edge cases.

## How and where has this been tested?

- **How this was tested:** 
- **Browser(s) and OS(s) this was tested with:** 

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [ ] If my change impacts documentation, I have updated the documentation accordingly.
- [ ] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] I have updated any relevant storybook stories and templates.
- [ ] If my change(s) include visual change(s), a designer has reviewed and approved those changes.

- [ ] This pull request is ready to merge.

### CODE REVIEW

**Comment by pfulton** [comment]
Released: `@spectrum-css/menu@5.0.0-beta.4`

File: components/menu/index.css [hunk 1]
```diff
@@ -10,170 +10,595 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+}
+
+.spectrum-Menu-chevron {
+  /* default: chevrons (aka collapsibles) that have no icon and instead have text directly next to them */
+  margin-inline-end: var(--mod-menu-item-text-to-control, var(--spectrum-menu-item-text-to-control));
+}
+
+/* chevrons (aka collapsibles) that have an icon directly next to them */
+.spectrum-Menu-chevron--withAdjacentIcon {
+  margin-inline-end: var(--mod-menu-item-text-to-control, var(--spectrum-menu-item-text-to-control));
```
Comment (najikahalsema) [inline components/menu/index.css:580]: If this value is the same as the value for the above `.spectrum-Menu-chevron`, couldn't we just get rid of this `--withAdjacentIcon` entirely? Same goes for `spectrum-Menu-checkmark--withAdjacentIcon`

Comment (jawinn) [reply components/menu/index.css:583]: 👍 I've removed these classes after double-checking with Patrick. They don't appear to be needed anymore.

**Comment by pfulton** [comment]
New beta published: `@spectrum-css/menu@5.0.0-beta.5`

**Comment by pfulton** [comment]
New beta published:  `@spectrum-css/menu@5.0.0-beta.6`

**Comment by pfulton** [comment]
New beta published: `@spectrum-css/menu@5.0.0-beta.7`

**Pull request merged**